### PR TITLE
KAFKA-6800: Update SASL/PLAIN and SCRAM docs to use KIP-86 callbacks

### PR DESCRIPTION
**Pull request by rajinisivaram**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by omkreddy** [comment]
LGTM

**Comment by rajinisivaram** [comment]
@junrao @ijuma Will one of you have time to review this docs PR, so that we can include in 2.0.0?

**Comment by ijuma** [comment]
Yes, I can review it today.

File: docs/security.html [hunk 1]
```diff
@@ -547,12 +547,20 @@ <h3><a id="security_sasl" href="#security_sasl">7.3 Authentication using SASL</a
...
-                <code>javax.security.auth.spi.LoginModule</code> that provides usernames and passwords from an external source. The login module implementation should
+                <a href="#security_sasl_plain_brokerconfig">here</a>. From Kafka version 2.0 onwards, you can avoid storing clear passwords on disk
+                by configuring your own callback handlers that obtain username and password from an external source using the configuration options
+                <code>sasl.server.callback.handler.class</code> and <code>sasl.client.callback.handler.class</code>.
+                <p>
+                With older versions of Kafka, you can plug in your own implementation of <code>javax.security.auth.spi.LoginModule</code>
+                that provides usernames and passwords from an external source. The login module implementation should
                 provide username as the public credential and password as the private credential of the <code>Subject</code>. The default implementation
-                <code>org.apache.kafka.common.security.plain.PlainLoginModule</code> can be used as an example.</li>
```
Comment (ijuma) [inline docs/security.html]: We normally just document the newest version since we keep the older version docs.

Comment (rajinisivaram) [reply docs/security.html]: @ijuma Thanks for the review, removed section for older versions.

**Review by ijuma — APPROVED** [review]
LGTM, thanks.

**Comment by rajinisivaram** [comment]
@ijuma Thanks for the review, merging to trunk.

**Pull request merged**